lib/wibox/widget/calendar.lua

```diff
@@ -0,0 +1,401 @@
+--
+-- @author getzze
+-- @copyright 2017 getzze
+-- @classmod wibox.widget.calendar
+---------------------------------------------------------------------------
+
+local setmetatable = setmetatable
+local string = string
+local gtable = require("gears.table")
+local vertical = require("wibox.layout.fixed").vertical
+local grid = require("wibox.layout.grid")
+local textbox = require("wibox.widget.textbox")
+local base = require("wibox.widget.base")
+local beautiful = require("beautiful")
+
+local calendar = { mt = {} }
+
+local properties = { "date", "font", "spacing", "hide_year", "week_numbers", "start_sunday", "fn_embed" }
+
+
+--- The calendar font.
+-- @beautiful beautiful.calendar_font
+-- @tparam string font Font of the calendar
+
+--- The calendar spacing.
+-- @beautiful beautiful.calendar_spacing
+-- @tparam number spacing Spacing of the grid (twice this value for inter-month spacing)
+
+--- Display the calendar week numbers.
+-- @beautiful beautiful.calendar_week_numbers
+-- @param boolean Display week numbers
+
+--- Start the week on Sunday.
+-- @beautiful beautiful.calendar_start_sunday
+-- @param boolean Start the week on Sunday
+
+--- The calendar date.
+--
+-- Date in space-separated integers format "[%day] [%month] %year".
+-- E.g.. "21 02 2005", "02 2005", "2005".
+-- @param string Date in space-separated integers format "[%day] [%month] %year".
+-- @property date
+
+--- The calendar font.
+--
+-- Choose a monospace font for a better rendering.
+--@DOC_wibox_widget_calendar_font_EXAMPLE@
+-- @param[opt="Monospace 10"] string Font of the calendar
+-- @property font
+
+--- The calendar spacing.
+--
+-- The spacing between cells in the month.
+-- The spacing between months in a year calendar is twice this value.
+-- @param[opt=5] number Spacing of the grid
+-- @property spacing
+
+--- Display the calendar week numbers.
+--
+--@DOC_wibox_widget_calendar_week_numbers_EXAMPLE@
+-- @param[opt=false] boolean Display week numbers
+-- @property week_numbers
+
+--- Start the week on Sunday.
+-- @param[opt=false] boolean Start the week on Sunday
+-- @property start_sunday
+
+--- Hide year number in month header.
+-- @param boolean Hide year in header (default: false for month calendar, true for year calendar)
+-- @property hide_year
+
+--- The widget encapsulating function.
+--
+-- Function that takes a widget and flag (`string`) as argument
+-- and returns a widget encapsulating the input widget.
+--
+-- Default value: function (widget, flag) return widget end
+--
+-- It is used to add a container to the grid layout and to the cells:
+--
+--@DOC_wibox_widget_calendar_fn_embed_cell_EXAMPLE@
+-- @param function Function to embed the widget depending on its flag
+-- @property fn_embed
+
+
+
+
+-- Parse a date for the day, month and year
+-- @tparam string date Date in the "%d %m %Y" format
+-- @treturn {number,number,number} List of day, month, year (nil if undefined)
+local function parse_date(date)
```

Comment:
`date` is a single property, otherwise there are three. but maybe a table would be ok.